visit-scheduler-api-specification.yaml

```diff
@@ -186,22 +193,24 @@ paths:
                 items:
                   $ref: '#/components/schemas/NonSuccessResponse'
         404:
-          description: Item not found for given input parameter
+          description: Visit Identifier not found
           content:
             application/json:
               schema:
                 items:
                   $ref: '#/components/schemas/NonSuccessResponse'
-  /visit-slot/{SlotReference}:
+  /visits-by-prisoner/{prisonerId}:
     get:
```

Comment:
Ditto again as per previous comment.

Reply:
By convention all urls in the Visits resource file would start with the same URL resource as the annotation in spring applies to the whole file. It would be awkward to try and have different resource names - _visits_ and _visits-by-prisoner_

Reply:
The `/visits` path could potentially get quite complicated to resolve:

We need to satisfy the following requests:

1. retrieve all visits for the given prisoner id
2. retrieve all visits for the given visitor id
3. retrieve all visits which are taking place today/this am/this pm -  for the given prison
4. retrieve all details for this visit id
5. probably a few others

Thats a fair few filters which I think will be used mutually exclusively - i.e I think we are unlikely to use more than one filter at a time.

I'm happy to load all of these onto `/visits` - but I thought separating them out as individual endpoints would make them easier to use - and code - maybe there is a better way to do this... - certainly one for discussion.

Reply:
I definitely (still) prefer 
/visits/prisoner/{prisonerId}  
/visits/id
/visits/visitor/{visitorId}
with some additional query params if necessary
The implementation and validation is clearer and less error prone
One for wednesday!